### ISSUE TITLE
div.modal-backdrop double times

### ISSUE THREAD

**Issue by laszlogmenyhart** [twbs/bootstrap#16581]
We have modal popup for confirming when user clicking on an item.
Sometimes(!) double-click occurs that two modal-backdrop div will be appended to the body.
Unfortunately only one of them will be removed after reading/clicking on the modal popup.

I found that 1100 line appends this new div to the body in file https://github.com/twbs/bootstrap/blob/master/dist/js/bootstrap.js.

I think it would be good in an "if" like this:

            if (this.$body.has("div.modal-backdrop").length == 0) {
                this.$backdrop = $('<div class="modal-backdrop ' + animate + '" />')
                    .appendTo(this.$body)
            }

Please advise!

Thanks!

**Comment by cvrebert** [comment]
@laszlogmenyhart Please post a JS Bin that demonstrates the problem.
Sounds similar to #15781.

**Comment by cvrebert** [comment]
And please state which OS and browser you're using.

**Comment by kkirsche** [comment]
@cvrebert should this have a js label?

**Comment by laszlogmenyhart** [comment]
Hi,

The problem was OS-independent and browser-independent.
It's hard to demonstrate this because we have event triggers and ajax calls and modals in a complex system.
But thanks this idea, because I found the problem meantime and I solved the issue. It happened when the server response arrived a bit slower.

The root cause was not because of bootstrap, but I will share a similar example with you when I will have time for this (after our project? :-( ).

So the modal window was shown two times, because both click event triggered an ajax call (first event did not hide the button in time), and both modal window was opened so two div.modal-backdrop was appended to the body. (I suggested to check whether it exists or not.) But the same container was used for both model so the second one overwrote the first one. Maybe we should use an identifier for these two different alert popups (modal windows) somehow. Just I did not know bootstrap modal window configuration enough deep. I did not find any queue-technic in bootstrap to present one after the other.

Thanks,

**Comment by kkirsche** [comment]
@laszlogmenyhart as you have solved your issue and you stated it was not bootstrap related I believe the issue should be closed.

**Comment by laszlogmenyhart** [comment]
Well, we use work-around (managing the server calls) here 
but
I think you should analyze this because it can happen later again for others. It seems to me (without low level source code analysis) that prototype.remoteBackdrop removes only (the last) one div, but prototype.backdrop appendTo body everytime when it called.

**Comment by mdo** [comment]
Closing since I think it was determined this isn't a bug. Holler if that's not the case.

**Comment by jamesmayes** [comment]
We encountered the same problem.  When a click and dblclick were triggered in rapid succession, the backdrop would be triggered twice.  Upon dismissal of the modal, only the last backdrop would be removed.  We are using bootstrap 2.1.x.  We decided to add an event handler for the hidden event and ensure that all .modal-backdrops were removed.  It is working well.  Bootstrap 3.x has a different event.
```
// for bootstrap 2.x, differs for 3.x
$modal.on('hidden', function () {
  $('.modal-backdrop').remove();
});
```
